### ISSUE TITLE
AMP cuda bfloat16?

### ISSUE THREAD

**Issue by yuyangxie96** [pytorch/pytorch#109099]
### 📚 The doc issue

In the link: [https://pytorch.org/docs/stable/amp.html?highlight=amp#module-torch.cuda.amp](https://pytorch.org/docs/stable/_modules/torch/cuda/amp/autocast_mode.html#autocast)
For CUDA and CPU, APIs are also provided separately:

torch.autocast("cuda", args...) is equivalent to torch.cuda.amp.autocast(args...).

torch.autocast("cpu", args...) is equivalent to torch.cpu.amp.autocast(args...). For CPU, only lower precision floating point datatype of torch.bfloat16 is supported for now.

[Autocasting](https://pytorch.org/docs/stable/amp.html?highlight=amp#autocasting)

[Gradient Scaling](https://pytorch.org/docs/stable/amp.html?highlight=amp#gradient-scaling)

[Autocast Op Reference](https://pytorch.org/docs/stable/amp.html?highlight=amp#autocast-op-reference)

[Op Eligibility](https://pytorch.org/docs/stable/amp.html?highlight=amp#op-eligibility)

[CUDA Op-Specific Behavior](https://pytorch.org/docs/stable/amp.html?highlight=amp#cuda-op-specific-behavior)

[CUDA Ops that can autocast to float16](https://pytorch.org/docs/stable/amp.html?highlight=amp#cuda-ops-that-can-autocast-to-float16)

[CUDA Ops that can autocast to float32](https://pytorch.org/docs/stable/amp.html?highlight=amp#cuda-ops-that-can-autocast-to-float32)

[CUDA Ops that promote to the widest input type](https://pytorch.org/docs/stable/amp.html?highlight=amp#cuda-ops-that-promote-to-the-widest-input-type)

[Prefer binary_cross_entropy_with_logits over binary_cross_entropy](https://pytorch.org/docs/stable/amp.html?highlight=amp#prefer-binary-cross-entropy-with-logits-over-binary-cross-entropy)

[CPU Op-Specific Behavior](https://pytorch.org/docs/stable/amp.html?highlight=amp#cpu-op-specific-behavior)

[CPU Ops that can autocast to bfloat16](https://pytorch.org/docs/stable/amp.html?highlight=amp#cpu-ops-that-can-autocast-to-bfloat16)

[CPU Ops that can autocast to float32](https://pytorch.org/docs/stable/amp.html?highlight=amp#cpu-ops-that-can-autocast-to-float32)

[CPU Ops that promote to the widest input type](https://pytorch.org/docs/stable/amp.html?highlight=amp#cpu-ops-that-promote-to-the-widest-input-type)

**So here is my question. Does torch.cuda.amp.autocast not support torch.bfloat16 training in GPU?**

**I try to run with the code:**
```
with torch.cuda.amp.autocast(enabled = True,dtype=torch.bfloat16):  
    logits, loss = classifier(X, Y) 
```

**the dtype of X or Y seems still in torch.float16 format.**
